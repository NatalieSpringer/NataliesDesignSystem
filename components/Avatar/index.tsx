import React, { useState } from "react";
import { View } from "react-native";
import { AvatarProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Image } from "expo-image";
import { Icon } from "../Icon";

export const Avatar = ({ size = "m", source }: AvatarProps) => {
	const { styles, theme } = useStyles(stylesheet, {
		size
	});

	const [loaded, setLoaded] = useState(false);

	const iconSize = styles.container.size;

	return (
		<View accessible accessibilityRole="image">
			<Image
				contentFit="cover"
				onLoad={() => setLoaded(true)}
				style={[styles.container, { position: "absolute" }]}
				source={source}
				testID="image"
				transition={800}
			/>
			{!loaded && (
				<View style={[styles.container]}>
					<Icon
						color={theme.avatar.icon}
						name="person"
						size={iconSize}
					/>
				</View>
			)}
		</View>
	);
};
