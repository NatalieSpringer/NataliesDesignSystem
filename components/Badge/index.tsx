import React from "react";
import { View } from "react-native";
import { BadgeProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";
import { Icon } from "../Icon";

export const Badge = ({ name, text, variant }: BadgeProps) => {
	const { styles, theme } = useStyles(stylesheet, {
		variant
	});

	const foregroundColor = styles.container.textColor;

	return (
		<View style={styles.container}>
			{!!text && (
				<Text color={foregroundColor} size="s" weight="bold">{text}</Text>
			)}
			{!!name && (
				<Icon color={foregroundColor} name={name} size={15} weight="bold" />
			)}
		</View>
	);
};
