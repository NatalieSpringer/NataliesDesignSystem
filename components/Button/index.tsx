import React from "react";
import { TouchableOpacity } from "react-native";
import { ButtonProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";

export const Button = ({
	disabled = false,
	onPress,
	text,
	variant
}: ButtonProps) => {
	const { styles } = useStyles(stylesheet, {
		variant
	});

	const containerStyle = styles.container(disabled);
	const { textColor } = containerStyle;

	return (
		<TouchableOpacity
			accessible
			accessibilityLabel={text}
			accessibilityRole="button"
			disabled={disabled}
			onPress={onPress}
			style={containerStyle}
		>
			<Text color={textColor} size="s" weight="bold">
				{text}
			</Text>
		</TouchableOpacity>
	);
};
