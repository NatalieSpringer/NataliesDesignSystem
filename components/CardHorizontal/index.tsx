import React from "react";
import { TouchableOpacity, View } from "react-native";
import { CardHorizontalProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";
import { Icon } from "../Icon";

export const CardHorizontal = ({
	description,
	onPress,
	title
}: CardHorizontalProps) => {
	const { styles, theme } = useStyles(stylesheet);

	return (
		<TouchableOpacity onPress={onPress} style={styles.container}>
			<View style={styles.textContainer}>
				<Text size="s" weight="bold">
					{title}
				</Text>
				<Text size="xs">{description}</Text>
			</View>
			<Icon color={theme.card.icon} name="chevronRight" size={16} />
		</TouchableOpacity>
	);
};
