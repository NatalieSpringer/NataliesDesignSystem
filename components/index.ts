export * from "./Avatar";
export * from "./Banner";
export * from "./Button";
export * from "./Dialog";
export * from "./Icon";
export * from "./Image";
export * from "./SegmentedControl";
export * from "./Text";
