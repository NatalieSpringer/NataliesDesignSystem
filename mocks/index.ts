import { UnistylesRuntime } from "react-native-unistyles";

export const mockAppTheme = (theme: "light" | "dark") => {
	// @ts-ignore
	jest.spyOn(UnistylesRuntime, "themeName", "get").mockReturnValueOnce(theme);
};
