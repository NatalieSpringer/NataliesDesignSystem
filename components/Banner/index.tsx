import React from "react";
import { View } from "react-native";
import { BannerProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Text } from "../Text";
import { Button } from "../Button";

export const Banner = ({ description, onPress, title }: BannerProps) => {
	const { styles } = useStyles(stylesheet);

	return (
		<View style={styles.container}>
			<Text color="black" size="s" weight="bold">
				{title}
			</Text>
			<Text color="black" size="xs">
				{description}
			</Text>
			<View style={styles.button}>
				<Button onPress={onPress} text="Button" variant="primary" />
			</View>
		</View>
	);
};
