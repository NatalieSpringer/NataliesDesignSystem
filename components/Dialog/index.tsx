import React from "react";
import { View } from "react-native";
import { DialogProps } from "./types";
import { stylesheet } from "./styles";
import { useStyles } from "react-native-unistyles";
import { Button } from "../Button";

export const Dialog = ({ buttons, variant }: DialogProps) => {
	const { styles } = useStyles(stylesheet, {
		variant
	});

	return (
		<View style={styles.container}>
			{variant === "primary" ? (
				<View style={styles.buttonContainer}>
					<View style={styles.primaryButton}>
						<Button {...buttons[0]} variant="secondary" />
					</View>
					<View style={styles.primaryButton}>
						<Button {...buttons[1]} variant="primary" />
					</View>
				</View>
			) : (
				<View style={styles.secondaryButton}>
					<Button {...buttons[0]} variant="secondary" />
					<Button {...buttons[1]} variant="secondary" />
					<Button {...buttons[2]} variant="primary" />
				</View>
			)}
		</View>
	);
};
