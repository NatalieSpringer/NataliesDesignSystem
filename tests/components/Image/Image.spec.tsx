import { Image } from "../../../components";
import { fireEvent, render } from "@testing-library/react-native";
import { mockAppTheme } from "../../../mocks";

describe("Image", () => {
	["dark", "light"].forEach((theme) => {
		["s", "m", "l", undefined].forEach((size) => {
			beforeEach(() => {
				mockAppTheme(theme as "light" | "dark");
			});
			it(`should render correctly ${size} in ${theme}`, () => {
				const tree = render(
					<Image
						size={size as "s" | "m" | "l"}
						source="https://picsum.photos/seed/696/3000/2000"
					/>
				).toJSON();

				expect(tree).toMatchSnapshot();
			});
		});
		it(`should render correctly with empty source`, () => {
			const tree = render(<Image size="s" />).toJSON();

			expect(tree).toMatchSnapshot();
		});
	});

	describe("integration", () => {
		it("should show image after onLoad is called", () => {
			const tree = render(
				<Image
					size={"s"}
					source="https://picsum.photos/seed/696/3000/2000"
				/>
			);

			const image = tree.getByTestId("image");
			fireEvent(image, "onLoad", { nativeEvent: {} });
		});
	});
});
