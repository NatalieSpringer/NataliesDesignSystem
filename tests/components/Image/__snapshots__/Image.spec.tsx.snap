// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render correctly l in dark 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4281282614}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={35}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#2F3036",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 35,
        "width": 190,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={190}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#2F3036",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 35,
          "width": 190,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 35,
              "width": 35,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly l in light 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4293587711}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={35}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#EAF2FF",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 35,
        "width": 190,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={190}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#EAF2FF",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 35,
          "width": 190,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 35,
              "width": 35,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly m in dark 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4281282614}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={30}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#2F3036",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 30,
        "width": 160,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={160}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#2F3036",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 30,
          "width": 160,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 30,
              "width": 30,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly m in light 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4293587711}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={30}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#EAF2FF",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 30,
        "width": 160,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={160}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#EAF2FF",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 30,
          "width": 160,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 30,
              "width": 30,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly s in dark 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4281282614}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={25}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#2F3036",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 25,
        "width": 130,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={130}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#2F3036",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 25,
          "width": 130,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 25,
              "width": 25,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly s in light 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4293587711}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={25}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#EAF2FF",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 25,
        "width": 130,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={130}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#EAF2FF",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 25,
          "width": 130,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 25,
              "width": 25,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly undefined in dark 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4281282614}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={30}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#2F3036",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 30,
        "width": 160,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={160}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#2F3036",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 30,
          "width": 160,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 30,
              "width": 30,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly undefined in light 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4281282614}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={30}
    source={
      [
        {
          "uri": "https://picsum.photos/seed/696/3000/2000",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#2F3036",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 30,
        "width": 160,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={160}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#2F3036",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 30,
          "width": 160,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 30,
              "width": 30,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly with empty source 1`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4293587711}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={25}
    source={
      [
        {
          "uri": "",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#EAF2FF",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 25,
        "width": 130,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={130}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#EAF2FF",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 25,
          "width": 130,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 25,
              "width": 25,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;

exports[`Image should render correctly with empty source 2`] = `
<View
  accessibilityRole="image"
  accessible={true}
>
  <ViewManagerAdapter_ExpoImage
    alignItems="center"
    aspectRatio={1.7777777777777777}
    backgroundColor={4281282614}
    borderRadius={10}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    justifyContent="center"
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    position="absolute"
    size={25}
    source={
      [
        {
          "uri": "",
        },
      ]
    }
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.7777777777777777,
        "backgroundColor": "#2F3036",
        "borderRadius": 10,
        "justifyContent": "center",
        "position": "absolute",
        "size": 25,
        "width": 130,
      }
    }
    testID="image"
    transition={
      {
        "duration": 800,
      }
    }
    width={130}
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "aspectRatio": 1.7777777777777777,
          "backgroundColor": "#2F3036",
          "borderRadius": 10,
          "justifyContent": "center",
          "size": 25,
          "width": 130,
        },
      ]
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="photo"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 25,
              "width": 25,
            },
            undefined,
          ],
        ]
      }
      tintColor="#B3DAFF"
      type="monochrome"
      weight="regular"
    />
  </View>
</View>
`;
